packages/react-native/src/hooks/useDamageImage/index.js

```diff
@@ -0,0 +1,78 @@
+import { useCallback, useMemo, useState } from 'react';
+import { Dimensions } from 'react-native';
+import PropTypes from 'prop-types';
+
+export default function useImageDamage(image) {
+  const width = Math.min(Dimensions.get('window').width - 50, 400);
+  const height = image.height * (width / image.width);
+
+  const [ellipseW, setEllipseW] = useState(null);
+  const [ellipseH, setEllipseH] = useState(null);
+  const [location, setLocation] = useState(null);
+  const [dragX, setDragX] = useState(false);
+  const [dragY, setDragY] = useState(false);
+  const [dragLocation, setDragLocation] = useState(false);
```

Comment:
Maybe a `useReducer` would has been better, but it's ok. Just read the doc about `useReducer`. https://fr.reactjs.org/docs/hooks-reference.html